src/protagonist/API/Converters/LegacyModeConverter.cs

```diff
@@ -0,0 +1,40 @@
+﻿using DLCS.Core;
+using DLCS.Core.Collections;
+using DLCS.HydraModel;
+
+namespace API.Converters;
+
+/// <summary>
+/// Converts legacy image API calls
+/// </summary>
+public static class LegacyModeConverter
+{
+    private const string DefaultMediaType = "dlcs/unknown";
+    
+    /// <summary>
+    /// Converts from legacy format to new format
+    /// </summary>
+    /// <param name="image">The image to convert</param>
+    /// <returns>A converted image</returns>
+    public static Image VerifyAndConvertToModernFormat(Image image)
+    {
+        if (image.MediaType is null)
+        {
+            var contentType = image.Origin?.Split('.').Last() ?? string.Empty;
+            
+            image.MediaType = MIMEHelper.GetContentTypeForExtension(contentType) ?? DefaultMediaType;
+            
+            if (image.Origin is not null && image.Family is null && image.DeliveryChannels.IsNullOrEmpty())
+            {
+                  image.Family = AssetFamily.Image;
+            }
+        }
+
+        if (image.MaxUnauthorised is null or 0)
+        {
+            image.MaxUnauthorised = -1;

```

Comment:
We only want to set to `-1` if Roles property is null or empty. (`Replace any 0 with -1 if roles is empty`).

The reason for this change is that the logic to work out if an asset requires auth has changed from `asset.MaxUnauth = 0 && hasRoles` to `asset.MaxUnauth == 0 || hasRoles`, so if a user provides `{ "maxUnauthorised": 0, "roles": "my-role" }` then that's a valid configuration to say that nothing should be displayed unless current user has `my-role`